staging/src/k8s.io/api/resource/v1alpha3/types.go

```diff
@@ -0,0 +1,998 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+const (
+	// Finalizer is the finalizer that gets set for claims
+	// which were allocated through a builtin controller.
+	// Reserved for use by Kubernetes, DRA driver controllers must
+	// use their own finalizer.
+	Finalizer = "dra.k8s.io/delete-protection"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:prerelease-lifecycle-gen:introduced=1.31
+
+// One or more slices represent a pool of devices managed by a given driver.
```

Comment:
How much do we want to follow godoc style for types.go? I was under the impression that this is primarily the input for user-facing documentation (e.g. via api/openapi-spec/swagger.json and eventually https://kubernetes.io/docs/reference/kubernetes-api/), so godoc is a secondary concern.

Having said that, for types it makes sense to follow godoc style because the documentation is still readable. For fields it becomes more awkward.

Reply:
Okay. Then let's also use `ResourceSlice` instead of just `slice` (like you did in your comment above). I picked the latter because it works in both kinds of docs, but it is less precise.

Reply:
I started replacing "claim" and "class" the same way, but then didn't make those changes because it became less readable. I think that is consistent with normal Go docs where "file" is a general concept and "os.File" a specific type.